src/Service.Host/client/src/components/parcels/ParcelsSearch.js

```diff
@@ -0,0 +1,329 @@
+
+    const handleParcelNumberSearchChange = (...args) => {
+        setParcelNumberSearch(args[1]);
+        setSearchTerm(
+            `${args[1]}&supplierIdSearchTerm=${supplierIdSearch}&carrierIdSearchTerm=${carrierIdSearch}&dateCreatedSearchTerm=${dateCreatedSearch}&supplierInvNoSearchTerm=${supplierInvNoSearch}&consignmentNoSearchTerm=${consignmentNoSearch}&commentsSearchTerm=${commentsSearch}`
+        );
+    };
+
+    const handleSupplierSearchChange = (...args) => {
+        setSupplierIdSearch(args[1]);
+        setSearchTerm(
+            `${parcelNumberSearch}&supplierIdSearchTerm=${args[1]}&carrierIdSearchTerm=${carrierIdSearch}&dateCreatedSearchTerm=${dateCreatedSearch}&supplierInvNoSearchTerm=${supplierInvNoSearch}&consignmentNoSearchTerm=${consignmentNoSearch}&commentsSearchTerm=${commentsSearch}`
+        );
+    };
+
+    const handleCarrierSearchChange = (...args) => {
+        setCarrierIdSearch(args[1]);
+        setSearchTerm(
+            `${parcelNumberSearch}&supplierIdSearchTerm=${supplierIdSearch}&carrierIdSearchTerm=${args[1]}&dateCreatedSearchTerm=${dateCreatedSearch}&supplierInvNoSearchTerm=${supplierInvNoSearch}&consignmentNoSearchTerm=${consignmentNoSearch}&commentsSearchTerm=${commentsSearch}`
+        );
+    };
+
+    const handleDateSearchChange = (...args) => {
+        setDateCreatedSearch(args[1]);
+        setSearchTerm(
+            `${parcelNumberSearch}&supplierIdSearchTerm=${supplierIdSearch}&carrierIdSearchTerm=${carrierIdSearch}&dateCreatedSearchTerm=${args[1]}&supplierInvNoSearchTerm=${supplierInvNoSearch}&consignmentNoSearchTerm=${consignmentNoSearch}&commentsSearchTerm=${commentsSearch}`
+        );
+    };
+
+    const handleSupplierInvSearchChange = (...args) => {
+        setSupplierInvNoSearch(args[1]);
+        setSearchTerm(
+            `${parcelNumberSearch}&supplierIdSearchTerm=${supplierIdSearch}&carrierIdSearchTerm=${carrierIdSearch}&dateCreatedSearchTerm=${dateCreatedSearch}&supplierInvNoSearchTerm=${args[1]}&consignmentNoSearchTerm=${consignmentNoSearch}&commentsSearchTerm=${commentsSearch}`
+        );
+    };
+
+    const handleConsignmentNoSearchChange = (...args) => {
+        setConsignmentNoSearch(args[1]);
+        setSearchTerm(
+            `${parcelNumberSearch}&supplierIdSearchTerm=${supplierIdSearch}&carrierIdSearchTerm=${carrierIdSearch}&dateCreatedSearchTerm=${dateCreatedSearch}&supplierInvNoSearchTerm=${supplierInvNoSearch}&consignmentNoSearchTerm=${args[1]}&commentsSearchTerm=${commentsSearch}`
+        );
+    };
+
+    const handleCommentSearchChange = (...args) => {
+        setCommentsSearch(args[1]);
+        setSearchTerm(
+            `${parcelNumberSearch}&supplierIdSearchTerm=${supplierIdSearch}&carrierIdSearchTerm=${carrierIdSearch}&dateCreatedSearchTerm=${dateCreatedSearch}&supplierInvNoSearchTerm=${supplierInvNoSearch}&consignmentNoSearchTerm=${consignmentNoSearch}&commentsSearchTerm=${args[1]}`
+        );
```

Comment:
These search strings are quite unwieldly but I'm not sure there's a better option..
You could store all the searchTerms in one state object

```
const [searchTerms, setSearchTerms] = useState({
    parcelNumberSearch: '',
    supplierIdSearch: '',
   etc... }
);
```

and then write a little function to build the big query strings by looping over the searchTerms keys and values, or something like that.

If it ends up being even messier then just leave it, but a little utility function (or hook?) to turn objects to searchTerms could be useful

Reply:
we also already have this package as a dependency that could help https://www.npmjs.com/package/query-string

Reply:
+1 query-string

Reply:
query string sounds good to me, I'll look it up properly and put it in